packages/core/src/components/axes/axis.ts

```diff
@@ -435,13 +450,27 @@ export class Axis extends Component {
 					.selectAll("g.tick text")
 					.data(label_data_array)
 					.text(function(d) {
-						return d.length > 18 ? d.substr(0, 8) + "..." + d.substr(-8) : d;
+						if (truncationTypeProvided === "TruncationTypes.MID_LINE") {
```

Comment:
this entire piece should go into `Tools.ts`, a function that'll take in text + a truncation type, and return truncated text